src/pass.h

```diff
@@ -206,11 +206,11 @@ struct PassOptions {
   //     cannot use a custom struct type as a function parameter type. Such a
   //     type would not be optimizable due to the constraint to not modify the
   //     type of the import/export, but it is very simple to use a type in a
-  //     single import/export and then all of its subtypes become unoptimizable,
-  //     which can mean all the types in your module if it is a root type such
-  //     as $java.lang.Object. For that reason we error on using such types on
-  //     the boundary for now. Instead, use basic types like anyref, externref,
-  //     etc.
+  //     single import/export and then all of its supertypes become
+  //     unoptimizable, which can mean all the types in your module if it is a
+  //     root type such as $java.lang.Object. For that reason we error on using
```

Comment:
The java.lang.Object example doesn't fit with the previous sentence because it doesn't have supertypes.

Reply:
Right, thanks, rewritten.